lib/Runtime/PlatformAgnostic/Platform/Linux/UnicodeText.ICU.cpp

```diff
@@ -42,6 +46,47 @@ namespace PlatformAgnostic
             return normalizer;
         }
 
+        static bool IsUtf16StringValid(const UChar* str, int* invalidIndex)
+        {
+            // ask ICU to scan through the string and see if it can be
+            // converted to utf8. If it can't, see if the error is that
+            // the string is malformed.
+
+            Assert(invalidIndex != nullptr);
+            *invalidIndex = -1;
+            int32_t length = u_strlen(str);
+            int32_t i = 0;
+
+            for (;;)
+            {
```

Comment:
1 - what happens here if str doesn't have 0 etc. and we end up with an undefined behavior?
2 - can we use size_t/unsigned instead of `int` - `int32_t` ? for `length`, and `i`, also could be nice for invalidIndex ( and reporting invalid index without -1, I don't think we need -1 there though )